app/tests/beaker-plugin-safe-app/test_modules/crypto_pub_enc_key.js

```diff
@@ -0,0 +1,43 @@
+let should = require('should');
+let testHelpers = require('./helpers');
+
+describe('window.safeCryptoPubEncKey', () => {
+  it('returns app\'s public encryption key as raw buffer', async () => {
+    const appHandle = await testHelpers.authoriseAndConnect();
+    const encKeyPairHandle = await window.safeCrypto.generateEncKeyPair(appHandle);
+    const pubEncKeyHandle = await window.safeCryptoKeyPair.getPubEncKey(encKeyPairHandle);
+    const rawKey = await window.safeCryptoPubEncKey.getRaw(pubEncKeyHandle);
+    should(rawKey.buffer.length).be.equal(32);
+  });
+
+  it('encrypts with a sealed box, using recipient\'s public key', async () => {
+    // NOTE: read about sealed boxes here: https://download.libsodium.org/doc/public-key_cryptography/sealed_boxes.html
+    const appHandle = await testHelpers.authoriseAndConnect();
+    const encKeyPairHandle = await window.safeCrypto.generateEncKeyPair(appHandle);
+    const pubEncKeyHandle = await window.safeCryptoKeyPair.getPubEncKey(encKeyPairHandle);
+    should(window.safeCryptoPubEncKey.encryptSealed(pubEncKeyHandle, 'encrypt this data'))
+    .be.fulfilled();
+  });
+
+  it.skip('encrypts with sender\'s secret key and recipient\'s public key', async () => {
+    const appHandle = await testHelpers.authoriseAndConnect();
+    const encKeyPairHandle = await window.safeCrypto.generateEncKeyPair(appHandle);
+    const pubEncKeyHandle = await window.safeCryptoKeyPair.getPubEncKey(encKeyPairHandle);
+    const secEncKeyHandle = await window.safeCryptoKeyPair.getSecEncKey(encKeyPairHandle);
+    const rawSecretKey = await window.safeCryptoSecEncKey.getRaw(secEncKeyHandle);
+    // evidently, rawSecretKey is the wrong argument to pass.
+    // QUESTION: a u64 should be passed to safe_client_libs, but i thought/
```

Comment:
@bochaco @krishnaIndia any light to shed on this? I'm not sure.

Reply:
There seems to be an issue there, even that the safe_app_nodejs (and its counterpart in the safe_app plugin) `PubEncKey.encrypt` documentation states that it expects either a buffer or string type for the secret key input param, it seems to be accepting only a `SecEncKey` object, therefore the test should provide the `secEncKeyHandle`, but first the safe_app plugin needs to be fixed as well as the documentation.